File: Java/Containers/MongoDB-REST/src/main/java/com/api/mongodb/rest/health/HealthResults.java
package com.api.mongodb.rest.health;

import java.util.List;

public class HealthResults {
    private String duration;
    private boolean healthy;
    private String status;
    private List<HealthResultEntry> results;
    private String timestamp;
    private String version;

    public HealthResults(long duration, List<HealthResultEntry> entries) {
        String status = "healthy";
        boolean healthy = true;
        for (HealthResultEntry entry : entries) {
            if (!entry.isHealthy()) {
                status = "unhealthy";
                healthy = false;
                break;
            }
        }

        this.duration = duration + " ms";
        this.healthy = healthy;
        this.status = status;
        this.results = entries;
        this.timestamp = new java.util.Date().toString();
        this.version = "1.0.0";
    }

    public String getDuration() {
        return duration;
    }

    public boolean isHealthy() {
        return healthy;
    }

    public String getStatus() {
        return status;
    }

    public List<HealthResultEntry> getResults() {
        return results;
    }

    public String getTimestamp() {
        return timestamp;
    }

    public String getVersion() {
        return version;
    }
}

File: Examples/PHP/Containers/Laravel-MongoDB-REST/src/app/Models/HealthResults.php
<?php

namespace App\Models;

class HealthResults implements \JsonSerializable
{
    private string $type;
    private string $duration;
    private bool $healthy;
    private string $status;
    private array $results;
    private string $timestamp;
    private string $version;

    public function __construct(int $duration, array $entries)
    {
        $this->type = "php";
        $this->duration = $duration . " ms";
        $this->healthy = true;
        foreach($entries as $entry) {
            if (!$entry->isHealthy()) {
                $this->healthy = false;
                break;
            }
        }
        $this->status = $this->healthy ? 'healthy' : 'unhealthy';
        $this->results = $entries;
        $this->timestamp = gmdate("Y-m-d\TH:i:s\Z");
        $this->version = "1.0.0";
    }

    public function getType(): string
    {
        return $this->type;
    }

    public function getDuration(): string
    {
        return $this->duration;
    }

    public function isHealthy(): bool
    {
        return $this->healthy;
    }

    public function getStatus(): string
    {
        return $this->status;
    }

    public function getResults(): array
    {
        return $this->results;
    }

    public function getTimestamp(): string
    {
        return $this->timestamp;
    }

    public function getVersion(): string
    {
        return $this->version;
    }

    public function jsonSerialize(): array
    {
        return [
            'type' => $this->type,
            'duration' => $this->duration,
            'healthy' => $this->healthy,
            'status' => $this->status,
            'results' => $this->results,
            'timestamp' => $this->timestamp,
            'version' => $this->version,
        ];
    }
}